tests/efs-check.sh

```diff
@@ -2,50 +2,45 @@
 
 set -e
 
-paths=${1}
+path=${1}
 fs_id=${2}
-paths=$(echo ${paths} | tr "," " ")
 status=0
 
-pathCount=$(cat /etc/fstab | grep -E "$(echo ${paths} | tr " " "|")" | wc -l | xargs)
-fsGrep=$(cat /etc/fstab | grep -E "fs-" | wc -l | xargs)
-truePathCount=$(echo ${paths} | tr " " "\n"| wc -l |xargs)
+path_exist=$(cat /etc/fstab | grep "$(echo ${path})";echo $?)
+fs_grep=$(cat /etc/fstab | grep "fs-";echo $?)
 
-if [ ${pathCount} == ${truePathCount} ] && [ ${fsGrep} == ${truePathCount} ]; then
-    echo "found the file system"
+if [[ ${path_exist} != 1 ]] && [[ ${fs_grep} != 1 ]]; then
+  echo "found the file system"
 else
-    echo "error: file system not found"
-    exit 1
+  echo "error: file system not found"
+  exit 1
 fi
 
 if [[ ${fs_id} ]]; then
-    fsCount=$(cat /etc/fstab | grep -E "${fs_id}" | wc -l | xargs)
-    if [[ ${fsCount} == "0" ]]; then
-        echo "Error: the given ${fs_id} was not attached"
-        exit 1
-    else
-        echo "Success: the given ${fs_id} is attached"
-    fi
+  fs_exist=$(cat /etc/fstab | grep "${fs_id}";echo $?)
+  if [[ ${fs_exist} == 1 ]]; then
+      echo "Error: the given ${fs_id} was not attached"
+      exit 1
+  else
+      echo "Success: the given ${fs_id} is attached"
+  fi
 fi
 
-computeIP=$(qhost | grep "ip-" | cut -d " " -f 1)
+compute_ip=$(qhost | grep "ip-" | cut -d " " -f 1)
 
-for path in ${paths}
-do
-    touch ${path}/master_file
-    scp ${path}/master_file ${computeIP}:${path}/compute_file
+touch ${path}/master_file
+scp ${path}/master_file ${compute_ip}:${path}/compute_file
 
-    computeCheck=$(ls ${path} | grep "compute_file")
+compute_check=$(ls ${path} | grep "compute_file";echo $?)
 
-    if [[ ${computeCheck} ]]; then
-        echo "${path} is correctly mounted on compute"
-        rm ${path}/compute_file
-    else
-        echo "error: ${path} is not found on compute"
-        status=1
-    fi
+if [[ ${compute_check} != 1 ]]; then
```

Comment:
Is the exit code guaranteed to 0 or 1?

Reply:
grep returns 1 if it cannot find the message and 0 if it can, but I check for != 1 because if grep finds a match the variable will be "matchstring"0, if it fails to find a match it would just be 1